src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -0,0 +1,175 @@
+                    return;
+
+                if (projectData.GetPropertyValue("OutputType") == executableProjectOutputType)
+                {
+                    if (ProjectHasUserCustomizedConfiguration(projectData))
+                    {
+                        shouldReturnNull = true;
+                        state.Stop(); // We don't want to override Configuration if ANY project in a sln uses a custom configuration
+                        return;
+                    }
+
+                    string useReleaseConfiguraton = projectData.GetPropertyValue(slnProjectConfigPropertytoCheck);
+                    if (!string.IsNullOrEmpty(useReleaseConfiguraton))
+                    {
+                        lock (projectDataLock)
+                        {
+                            configuredProjects.Add(projectData); // we don't care about race conditions here
+                            configValues.Add(useReleaseConfiguraton.ToLower());
+                        }
+                    }
+                }
+            });
+
+            if (configuredProjects.Any() && configValues.Count > 1 && !shouldReturnNull)
+            {
+                throw new GracefulException(CommonLocalizableStrings.SolutionExecutableConfigurationMismatchError, slnProjectConfigPropertytoCheck, String.Join("\n", (configuredProjects).Select(x => x.FullPath)));
+            }
+
+            return shouldReturnNull ? null : configuredProjects.FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Returns dotnet CLI command-line parameters (or an empty list) to change configuration based on 
+        /// a boolean that may or may not exist in the targeted project.
+        /// <param name="defaultedConfigurationProperty">The boolean property to check the project for. Ex: PublishRelease</param>
+        /// <param name="slnOrProjectArgs">The arguments or solution passed to a dotnet invocation.</param>
+        /// <param name="configOption">The arguments passed to a dotnet invocation related to Configuration.</param>
+        /// </summary>
+        /// <returns>Returns a string such as -property:configuration=value for a projects desired config. May be empty string.</returns>
+        public IEnumerable<string> GetCustomDefaultConfigurationValueIfSpecified(
+            ParseResult parseResult,
+            string defaultedConfigurationProperty,
+            IEnumerable<string> slnOrProjectArgs,
+            Option<string> configOption,
+            IEnumerable<string> userPropertyArgs
+            )
+        {
+            ProjectInstance project = null;
+
+            IEnumerable<string> calledArguments = parseResult.Tokens.Select(x => x.ToString());
+            IEnumerable<string> slnProjectAndCommandArgs = slnOrProjectArgs.Concat(calledArguments);
+            if (!parseResult.HasOption(configOption) && !ArgsContainsProperty(userPropertyArgs, "Configuration") && !ArgsContainsProperty(userPropertyArgs, defaultedConfigurationProperty))
```

Comment:
Should we consider placing the Configuration and Release values in a const or readonly string in case they are needed elsewhere to avoid any typos in the future?

Reply:
Thats a good idea. I could move the `PackRelease` and `PublishRelease` const there too. Where would be a good place, shall we make a `\src\Common\PropertyNames.cs?`

Reply:
Suggestion applied